File: src/main/java/burp/model/ParameterMap.java
package burp.model;

import java.util.List;

import org.apache.jena.rdf.model.RDFNode;

import burp.vocabularies.RML;

public class ParameterMap extends TermMap {

	public ParameterMap(Expression expression) {
		super(expression);
        termType = RML.IRI;
	}
	
	@Override
	public List<RDFNode> generateTerms(Iteration i, String baseIRI) {
		if(termType == RML.IRI) {
			return generateIRIs(i, baseIRI);			
		}
		
		throw new RuntimeException("Incorrect term type for parameter map.");	
	}

	@Override
	public boolean isGatherMap() {
		return false;
	}
	
}

File: src/main/java/burp/model/ParameterMap.kt
package burp.model

import burp.vocabularies.RML
import org.apache.jena.rdf.model.Resource

class ParameterMap : TermMap() {
    init {
        termType = RML.IRI
    }

    override fun getName() = "parameter map"

    override fun getAllowedTermTypes(): Set<Resource> = setOf(RML.IRI, RML.URI)
}